src/components/store/reducers/cart.ts

```diff
@@ -0,0 +1,91 @@
+import {createSlice, PayloadAction} from '@reduxjs/toolkit';
+import {Product} from '../../types/product';
+import {CartItemType} from "../../types/cartItemType";
+import {loadState} from "./storeDb";
+import {CartItemArgType} from "../../types/cartItemArgType";
+
+/**
+ * начальное состояние стейта
+ */
+let initialState = loadState<CartItemType[]>('cart', []);
+
+const slice = createSlice({
+  name: 'cart',
+  initialState,
+  reducers: {
+    /**
+     * добавляет продукт в стейт корзины, при повторном добавлении увеличивает quantity
+     */
+    addProductToCart: (state, { payload: cartItem }: PayloadAction<CartItemArgType>): CartItemType[] => {
+      const item = state.find((item) => item.product.id === cartItem.product.id);
+      if (!item) {
+        return [...state, {
+          id: state.length + 1,
+          product: cartItem.product,
+          quantity: cartItem.quantity
+        }];
+      }
+      item.quantity += cartItem.quantity;
```

Comment:
Do not change `state` directly.
I think, you need to return `[...state, ...our updates]`